xpdacq/xpdacq.py

```diff
@@ -192,6 +192,58 @@ def _parse_calibration_file(config_file_name):
                 config_dict[option] = None
     return config_dict
 
+def new_prun(sample, scanplan, auto_dark = glbl.auto_dark, **kwargs):
+    '''on this 'sample' run this 'scanplan'
+        
+    Arguments:
+    sample - sample metadata object
+    scanplan - scanplan metadata object
+    auto_dark - optional. Type auto_dark = False to suppress the automatic collection of a dark image (default = True). Strongly recommend to leave as True unless problems are encountered
+    **kwargs - dictionary that will be passed through to the run-engine metadata
+    '''
+    scan = Scan(sample, scanplan)
+    scan.md.update({'sc_isprun':True})
+    if auto_dark:
+        auto_dark_md_dict = _auto_dark(scan)
```

Comment:
Yes, I am moving toward this direction. My goal of this PR is to have `execute_scans()` which includes automated logics (dark and load calibration parameters) and other necessary logics. `dark`, `background`...e.t.c  will all call `execute_scans()`.  The final code could look like 
```
def prun(sample,  scanplan, <options>):
    scan = Scan(sample, scanplan)
    scan.md.update({'is_prun':True})
    _execute_scans(scan, <options>)
    return

def dark(sample,  scanplan, <options>):
    scan = Scan(sample, scanplan)
    scan.md.update({'is_dark':True})
    _execute_scans(scan, <options>)
    return
...
```